tests/parse_config_update_test.py

```diff
@@ -170,3 +173,53 @@ def test_parse_streams_skips_stream_info_if_remove_config_and_schema_present_but
     config_message = deserialise_rf5k(message)
     streams = tuple(_parse_streams(CommandType.REMOVE, config_message.streams))
     assert not streams
+
+
+@patch(
+    "forwarder.parse_config_update.deserialise_rf5k",
+    side_effect=RuntimeError("Runtime Error"),
+)
+def test_parse_config_update_when_runtime_error_occurs(mock_func):
+    test_channel_name = "test_channel"
+    message = serialise_rf5k(
+        UpdateType.ADD,
+        [StreamInfo(test_channel_name, "f142", "output_topic", Protocol.PVA)],
+    )
+    config_update = parse_config_update(message)
+    assert config_update.command_type == CommandType.INVALID
+
+
+@patch(
+    "forwarder.parse_config_update.deserialise_rf5k",
+    side_effect=flatbuffer_struct.error("Flatbuffer Error"),
+)
+def test_parse_config_update_when_flatbuffer_struct_error_occurs(mock_func):
+    test_channel_name = "test_channel"
+    message = serialise_rf5k(
+        UpdateType.ADD,
+        [StreamInfo(test_channel_name, "f142", "output_topic", Protocol.PVA)],
+    )
+    config_update = parse_config_update(message)
+    assert config_update.command_type == CommandType.INVALID
+
+
+def test_parse_config_update_when_invalid_update_type_is_provided():
+    invalid_update_type = UpdateType.REMOVEALL + 1
+    test_channel_name = "test_channel"
+    message = serialise_rf5k(
+        invalid_update_type,
+        [StreamInfo(test_channel_name, "f142", "output_topic", Protocol.PVA)],
+    )
+    config_update = parse_config_update(message)
+    assert config_update.command_type == CommandType.INVALID
+
+
+def test_parse_streams_when_invalid_protocol_is_provided():
+    invalid_protocol = Protocol.FAKE + 1
```

Comment:
This test will start failing if we add a new protocol and suddenly Protocol.FAKE + 1 is valid.
Same for some of the other tests above.

Perhaps we can use negative values instead, if that is allowed by the underlying library.
If that is not possible then a much bigger number will have to do.

Reply:
Negative are not possible so I put 9999 (clear indication hopefully that it is invalid and should be large enough).